datafusion/functions/src/string/repeat.rs

```diff
@@ -65,10 +65,17 @@ impl Default for RepeatFunc {
 impl RepeatFunc {
     pub fn new() -> Self {
         Self {
-            signature: Signature::coercible(
+            signature: Signature::coercible_v2(
                 vec![
-                    TypeSignatureClass::Native(logical_string()),
-                    TypeSignatureClass::Native(logical_int64()),
```

Comment:
In v1 version, any integer is casted to the defined NativeType's default casted type which is i64 in this case.

Reply:
I do like how much more explicit the new formualtion is